crates/fuel-gas-price-algorithm/src/lib.rs

```diff
@@ -4,3 +4,24 @@
 
 pub mod v0;
 pub mod v1;
+
+#[allow(clippy::cast_possible_truncation)]
+pub(crate) fn cumulative_percentage_change(
+    new_exec_price: u64,
+    for_height: u32,
+    percentage: u64,
+    height: u32,
+) -> u64 {
+    let blocks = height.saturating_sub(for_height) as f64;
+    let percentage_as_decimal = percentage as f64 / 100.0;
+    let multiple = (1.0f64 + percentage_as_decimal).powf(blocks);
+    let mut approx = new_exec_price as f64 * multiple;
+    // account for rounding errors and take a slightly higher value
+    const ARB_CUTOFF: f64 = 16948547188989277.0;
+    if approx > ARB_CUTOFF {
+        const ARB_ADDITION: f64 = 2000.0;
+        approx += ARB_ADDITION;
+    }
```

Comment:
I don't understand what happens here. What's the meaning of `ARB_CUTOFF`?

Reply:
It's a hack for dealing with really large floats. The rounding error will be on the order of `2000` and based on the prop tests will always result in a value that is larger than the `actual` worst case value. 

So `ARB_CUTOFF` is where we start seeing the error and `ARB_ADDITION` is the the amount we add to account for that error. This is the cost of using floats to represent these values, maybe there is a cleaner way to represent it :).

Reply:
It doesn't sound arbitrary then. Can we rename it to something like this?
```suggestion
    // account for rounding errors and take a slightly higher value
    const ROUNDING_ERROR_CUTOFF: f64 = 16948547188989277.0;
    if approx > ROUNDING_ERROR_CUTOFF {
        const ROUNDING_ERROR_COMPENSATION: f64 = 2000.0;
        approx += ROUNDING_ERROR_COMPENSATION;
    }
```

Reply:
Background:

https://github.com/FuelLabs/fuel-core/pull/2025#discussion_r1681725245

Reply:
> It doesn't sound arbitrary then. Can we rename it to something like this?

There isn't a "correct" value to choose here, that's why I called it `ARB`, but you're right that the reason we chose it isn't completely arbitrary and your suggestion is much clearer and helpful anyway.

Reply:
Variable names updated in https://github.com/FuelLabs/fuel-core/pull/2219/commits/4337b14502861275c52aabedc4d9396b6f9e5147